src/telliot_feed_examples/reporters/reporter_autopay_utils.py

```diff
@@ -0,0 +1,226 @@
+import math
+from typing import Optional
+from typing import Tuple
+
+from eth_abi import decode_single
+from telliot_core.data.query_catalog import query_catalog
+from telliot_core.tellor.tellorflex.autopay import TellorFlexAutopayContract
+from telliot_core.utils.response import error_status
+from telliot_core.utils.timestamp import TimeStamp
+
+from telliot_feed_examples.feeds import CATALOG_FEEDS
+from telliot_feed_examples.utils.log import get_logger
+
+
+logger = get_logger(__name__)
+
+# List of currently active queries
+query_ids_in_catalog = {
+    query_catalog._entries[tag].query_id: tag for tag in query_catalog._entries
+}
+
+
+# Mapping to feed details
+class DetailsMap:
+    reward = 0
+    balance = 1
+    startTime = 2
+    interval = 3
+    window = 4
+    priceThreshold = 5
+    feedsWithFundingIndex = 6
+
+
+detail = DetailsMap()
+
+
+async def single_tip_suggested_query_id(
+    autopay: TellorFlexAutopayContract,
+) -> Tuple[Optional[str], Optional[int]]:
+    """Returns the currently suggested query to for a single tip report against.
+
+    Pulls query_ids with tips available from the tellor autopay
+    contract, checks if query ids exist in the query catalog,
+    then sorts them from highest tip amount to lowest
+    """
+
+    assert isinstance(autopay, TellorFlexAutopayContract)
```

Comment:
Instead of a bunch of assert statements, which will halt telliot from running, I'd check if the given contract is connected or if the variable is not None, or catch and log exceptions